tests/translator/input/function_with_custom_conditional_codedeploy_deployment_preference.yml

```diff
@@ -0,0 +1,17 @@
+AWSTemplateFormatVersion : '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Parameters:
+  EnvType:
+    Default: dev
+Conditions:
+  IsDevEnv: !Equals [!Ref EnvType, dev]
+  IsDevEnv2: !Equals [!Ref EnvType, prod]
+Resources:
+  HelloWorldFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.handler
+      Runtime: nodejs8.10
+      AutoPublishAlias: live
+      DeploymentPreference:
+        Type: !If [IsDevEnv, !If [IsDevEnv2, AllAtOnce, TestCustomDeploymentConfig], Canary10Percent15Minutes]
```

Comment:
Thanks for adding this test! Can you also add an example using `!Sub` or `!Join` to test intrinsic functions that don't follow the same format as `!If`?

Reply:
Nice!